packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart

```diff
@@ -0,0 +1,678 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import '../app_bar.dart';
+import '../back_button.dart';
+import '../button_bar.dart';
+import '../button_theme.dart';
+import '../color_scheme.dart';
+import '../debug.dart';
+import '../dialog.dart';
+import '../flat_button.dart';
+import '../icon_button.dart';
+import '../icons.dart';
+import '../material_localizations.dart';
+import '../scaffold.dart';
+import '../text_theme.dart';
+import '../theme.dart';
+
+import 'calendar_date_range_picker.dart';
+import 'date_picker_common.dart';
+import 'date_picker_header.dart';
+import 'date_utils.dart' as utils;
+import 'input_date_range_picker.dart';
+
+const Size _inputPortraitDialogSize = Size(330.0, 270.0);
+const Size _inputLandscapeDialogSize = Size(496, 164.0);
+const Duration _dialogSizeAnimationDuration = Duration(milliseconds: 200);
+const double _inputFormPortraitHeight = 98.0;
+const double _inputFormLandscapeHeight = 108.0;
+
+/// Shows a full screen modal dialog containing a material design date range picker.
+///
+/// The returned [Future] resolves to the [DateTimeRange] selected by the user when
+/// the user saves their selection. If the user cancels the modal, null is
+/// returned.
+///
+/// If [initialDateRange] is non-null, then it will be used as the initially selected
+/// date range. If it is provided, [initialDateRange.start] must be before or on
+/// [initialDateRange.end].
+///
+/// The [firstDate] is the earliest allowable date. The [lastDate] is the latest
+/// allowable date. Neither can be null.
+///
+/// If an initial date range is provided, [initialDateRange.start]
+/// and [initialDateRange.end] must both fall between or on [firstDate] and [lastDate].
+/// For all of these [DateTime] values, only their dates are considered. Their
+/// time fields are ignored.
+///
+/// The [currentDate] represents the current day (i.e. today). This
+/// date will be highlighted in the day grid. If null, the date of
+/// `DateTime.now()` will be used.
+///
+/// An optional [initialEntryMode] argument can be used to display the date
+/// picker in the [DatePickerEntryMode.calendar] (a scrollable calendar month grid)
+/// or [DatePickerEntryMode.input] (two text input fields) mode.
+/// It defaults to [DatePickerEntryMode.calendar] and must be non-null.
+///
+/// The following optional string parameters allow you to override the default
+/// text used for various parts of the dialog:
+///
+///   * [helpText], label displayed at the top of the dialog.
```

Comment:
I'd add definite articles here and below:
```suggestion
///   * [helpText], the label displayed at the top of the dialog.
```